seqr/views/utils/airflow_utils.py

```diff
@@ -0,0 +1,118 @@
+import json
+from google.auth.transport.requests import Request
+from google.oauth2 import id_token
+import requests
+
+from seqr.utils.communication_utils import safe_post_to_slack
+from seqr.utils.logging_utils import SeqrLogger
+from settings import AIRFLOW_API_AUDIENCE, AIRFLOW_WEBSERVER_URL, AIRFLOW_DAG_VERSION, \
+    SEQR_SLACK_LOADING_NOTIFICATION_CHANNEL
+
+logger = SeqrLogger(__name__)
+
+
+class DagRunningException(Exception):
+    pass
+
+
+def trigger_data_loading(dag_name, projects, data_path, additional_dag_variables, user,
+                         success_message, success_slack_channel, error_message):
+    success = True
+    updated_variables = _construct_dag_variables(projects, data_path, additional_dag_variables)
+    dag_id = f'seqr_vcf_to_es_{dag_name}_v{AIRFLOW_DAG_VERSION}'
+
+    try:
+        _check_dag_running_state(dag_id)
+        _update_variables(dag_name, updated_variables)
+        _wait_for_dag_variable_update(dag_id, projects)
+        _trigger_dag(dag_id)
+    except Exception as e:
+        logger_call = logger.warning if isinstance(e, DagRunningException) else logger.error
+        logger_call(str(e), user)
+        _send_slack_msg_on_failure_trigger(e, dag_id, updated_variables, error_message)
+        success = False
+
+    if success or success_slack_channel != SEQR_SLACK_LOADING_NOTIFICATION_CHANNEL:
+        _send_load_data_slack_msg(success_message, success_slack_channel, dag_id, updated_variables)
```

Comment:
The success channel could be anything we want, we currently use this function twice but we could easily use it a third time or more all with a different channel. `no_notification` is inaccurate - what this behavior is doing is saying if success and error notifications go to the same channel than you should just send one notification to that channel nstead of 2. I like the current behavior because it is explicit and maintainable - you pass one argument in which is the notification channel, and then the function logic handles any edge cases